c2_utils/src/mfx_va_frame_pool_allocator.cpp

```diff
@@ -88,87 +88,14 @@ mfxStatus MfxVaFramePoolAllocator::AllocFrames(mfxFrameAllocRequest *request,
             }
 
             response->NumFrameActual = 0;
-#define RETRY_TIMES 5
-            for (int i = 0; i < opt_buffers; ++i) {
```

Comment:
Could you explain why this pre-allocation can cause video stuck issue? How much time does it take for 4k buffer allocation?

Reply:
> Could you explain why this pre-allocation can cause video stuck issue? How much time does it take for 4k buffer allocation?

From android 15, Google introduced the GraphicsTracker to track the BufferPool usage. For example, if we set BP count to 25, we can only fetch less than or equal to 25 buffers. But when resolution is changed, we will hold current surface and re-init decoder for 25 buffers allocation, this is over 25 buffers to be needed. It will stuck in calling fetchGraphicBlock function until timeout(about 10sec), and fetchGraphicBlock will never succeed unless the surface release.